src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/SimpleRecipeHandlerSecondaryOutput.java

```diff
@@ -0,0 +1,119 @@
+        @Example(".input(item('minecraft:gold_block')).output(item('minecraft:emerald'), item('minecraft:nether_star')).secondaryChance(0.25).time(50)")
+    })
+    public SimpleRecipeHandlerSecondaryOutput.RecipeBuilder recipeBuilder() {
+        return new SimpleRecipeHandlerSecondaryOutput.RecipeBuilder();
+    }
+
+    SimpleRecipeHandlerSecondaryOutput(String name, SimpleRecipeManagerSecondaryOutput instance) {
+        super(name, instance);
+    }
+
+    @Property(property = "input", valid = @Comp("1"))
+    @Property(property = "output", valid = {@Comp(value = "1", type = Comp.Type.GTE), @Comp(value = "2", type = Comp.Type.LTE)})
+    public class RecipeBuilder extends AbstractRecipeBuilder<SimpleRecipe> {
+        @Property(valid = @Comp(value = "1", type = Comp.Type.GTE))
+        private int time = -1;
+
+        @Property(valid = @Comp(value = "1", type = Comp.Type.LTE))
+        private float secondaryChance = 1.0f;
+
+        @RecipeBuilderMethodDescription
+        public SimpleRecipeHandlerSecondaryOutput.RecipeBuilder time(int time) {
+            this.time = time;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public SimpleRecipeHandlerSecondaryOutput.RecipeBuilder secondaryChance(float secondaryOutput) {
+            this.secondaryChance = secondaryOutput;
+            return this;
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 1, 1, 1, 2);
+            validateFluids(msg);
+            if (time == -1) {
+                time = getDefaultTime();
+            }
+            msg.add(secondaryChance > 1.0f || secondaryChance < 0.0f, "secondary output has to be between 0 and 1, got {}", secondaryChance);
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return String.format("Error adding ProdigyTech %s Recipe", name);
+        }
+
+        @Override
+        @RecipeBuilderRegistrationMethod
+        public @Nullable SimpleRecipeSecondaryOutput register() {
+            if (!validate()) return null;
+            SimpleRecipeSecondaryOutput recipe = null;
+            IIngredient input1 = input.get(0);
+            ItemStack secondaryOutput = output.size() == 1 ? ItemStack.EMPTY : output.get(1);
+            if (input1 instanceof OreDictIngredient) {
+                recipe = new SimpleRecipeSecondaryOutput(((OreDictIngredient) input1).getOreDict(), output.get(0), secondaryOutput, time, secondaryChance);
+                addRecipe(recipe);
+            } else {
+                for (ItemStack input : input1.getMatchingStacks()) {
+                    recipe = new SimpleRecipeSecondaryOutput(input, output.get(0), secondaryOutput, time, secondaryChance);
+                    addRecipe(recipe);
+                }
+            }
+            return recipe;
+        }
+    }
+}
+
+@RegistryDescription
+class HeatSawmill extends SimpleRecipeHandlerSecondaryOutput {

```

Comment:
could you move this to be a public inner class? same logic as in the other location.